mbs_results/pivot_imputation_value.py

```diff
@@ -0,0 +1,169 @@
+    count_cell : str
+        name of column in count_df dataframe containing cell variable
+    input_date : str
+        name of column in input_df dataframe containing date variable
+    count_date : str
+        name of column in count_df dataframe containing date variable
+    identifier : str
+        name of column in input_df containing identifier variable
+
+    Returns
+    -------
+    Dataframe resulting from the left-join of input_df and count_df on the cell and
+    date columns.
+    """
+    df_merge = pd.merge(
+        input_df,
+        count_df,
+        how="left",
+        left_on=[input_cell, input_date],
+        right_on=[count_cell, count_date],
+    ).astype({identifier: "int"})
+
+    return df_merge.drop(columns=[count_cell, count_date])
+
+
+def pivot_imputation_value(
+    df: pd.DataFrame,
+    identifier: str,
+    date: str,
+    sic: str,
+    cell: str,
+    forward: str,
+    backward: str,
+    construction: str,
+    question: str,
+    imputed_value: str,
+    f_count: str,
+    b_count: str,
+    c_count: str,
+    selected_periods: list = None,
+) -> pd.DataFrame:
+
+    """
+    Returning dataframe containing imputation_value, filtered by date, pivoted by
+    imputation type and grouped by sic, cell, question and imputation type.
+
+    Parameters
+    ----------
+    dataframe : pd.DataFrame
+        Reference dataframe containing links, count values, and imputed values
+        by identifier, cell, date, and question
+    identifier : str
+        name of column in dataframe containing identifier variable
+    date : str
+        name of column in dataframe containing date variable
+    sic : str
+        name of column in dataframe containing sic variable
+    cell : str
+        name of column in dataframe containing cell variable
+    forward : str
+        name of column in dataframe containing forward link variable
+    backward : str
+        name of column in dataframe containing backward link variable
+    construction : str
+        name of column in dataframe containing construction link variable
+    question : str
+        name of column in dataframe containing question code variable
+    imputed_value: str
+        name of column in dataframe containing imputed_value variable
+    f_count: str,
+        name of column in dataframe containing f_count variable
+    b_count: str,
+        name of column in dataframe containing b_count variable
+    c_count: str,
+        name of column in dataframe containing c_count variable
+    selected_periods: list,
+        list containing periods to include in output
+
+    Returns
+    -------
+    dataframe filtered by date, containing imputation_value, pivoted by imputation type
+    and grouped by sic, cell, question and imputation type.
+
+    """
+    if selected_periods is not None:
+        df = df.query("{} in {}".format(date, selected_periods))
+
+    links_df = df.melt(
+        id_vars=[date, sic, cell, question, imputed_value],
```

Comment:
why imputed_value needs to be in index ?

Reply:
Am I right in thinking we are going to keep this in (following discussion on Teams)?